### DISCUSSION TITLE
how to retrieve socket ref by sid

### DISCUSSION THREAD

**Discussion by tambietgaumisanhe** [Totodore/socketioxide#214] [Q&A] [answered]
i want to pickup socket_ref by sid to emit event but i don't find any suitable method

**Comment by Totodore** [comment]
You currently can't, your only solution is to make the socket join a room corresponding to its id at connection (it is what is done internally with socket.io).
I will create a tracking issue for this feature.

**Reply by tausifcreates** [reply]
@Totodore As far as I know, in the latest Socket IO they don't automatically connect the socket to a room based on its id. You have to manually do this.

**Reply by Totodore** [reply]
Indeed it is only in V3 :)

**Reply by tambietgaumisanhe** [reply]
currently i'm using
` [🔌protocol v4](https://github.com/socketio/socket.io-protocol/tree/v4) : based on engine.io v3, under the feature flag v4, (socket.io js from v1.0.3..latest). `
Is it possible to find socket by sid?

**Reply by Totodore** [reply]
Unfortunately no, you have to use the solution I gave above.

**Reply by tambietgaumisanhe** [reply]
tyvm

**Reply by Totodore** [reply]
@fundon made a PR for this feature. A version with the `get_socket` fn will be released today for v0.8 and v0.9 :)

**Reply by Totodore** [reply, marked as answer]
Starting from 0.9.1: https://docs.rs/socketioxide/latest/socketioxide/struct.SocketIo.html#method.get_socket
